### ISSUE TITLE
next-translate-plugin breaks SSG in app router

### ISSUE THREAD

**Issue by jakubjanousek** [aralroca/next-translate#1154]
**What version of this package are you using?**
2.6.0

**What operating system, Node.js, and npm version?**
node v20.4.0

**What happened?**
Once `next-translate-plugin` is used in `next.config.js`, pages cannot be statically generated in app router.  There is nothing dynamic on the page or layout

`module.exports = nextTranslate();` results in `┌ λ /                                    5.35 kB        85.8 kB`
`module.exports = {}` results in `┌ ○ /                                    5.25 kB        85.6 kB`

```
λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
```

i18n.json is
```
{
  "locales": ["en"],
  "defaultLocale": "en",
  "pages": {
    "*": ["common"]
  }
}
```

Any ideas why this is happening? You can check a minimal example at https://github.com/jakubjanousek/next-translate-ssg-issue

**What did you expect to happen?**
Pages without getServerSideProps will be statically generated

**Are you willing to submit a pull request to fix this bug?**

**Comment by hensansi** [comment]
@aralroca I found the same issue recently when I started to migrate pages from `pages` to `app`. The first page I migrated was a static page so I simply used a `export const dynamic = "force-static";` to deal with it.

Right now I started to migrate pages with translations and this is definitely a bigger issue given that any page I migrate will be dynamic because of next-translate. Are we missing something with the next-translate configuration?

**Comment by jakubjanousek** [comment]
good catch @hensansi, unfortunately I am not familiar with the reasons why searchParams is there

for now, we have removed `next-translate-plugin` and created custom `useTranslation` that just directly calls transCore with statically imported translation file

this allows us to have SSG by default while keeping all of the next-translate interface, but it is a very specific use case, since each of our languages is a single site that knows its locale at build time

**Comment by kkak10** [comment]
I'm having the same problem

@jakubjanousek Can I share your `useTranslation` implementation code?

**Comment by astorozhevsky** [comment]
> for now, we have removed next-translate-plugin and created custom useTranslation that just directly calls transCore with statically imported translation file

@jakubjanousek Could you please share the code snippet or implementation details for this? It would be greatly helpful to understand how you integrated this into your setup.

**Comment by jakubjanousek** [comment]
Our situation is a bit specific that we have several deployment, but each has only a single translation. Therefore we are able to copy only the necessary translation before starting the server into current

a simplified version is something like this

```
import transCore from 'next-translate/transCore'
import currentLocale from 'locales/current/main.json'
import { Translate } from 'next-translate'

const lang = process.env.LOCALE || 'en'

const tWithoutNs = transCore({
  lang,
  config: {
    locales: [locale],
    defaultLocale: locale,
    defaultNS: 'main',
    loader: false,
    pages: {
      '*': ['main'],
    },
  },
  pluralRules: new Intl.PluralRules(lang),
  allNamespaces: { main: currentLocale as any },
})

export const getTranslate = () =>
  ({
    t: (key: string, query?: any, options = {}) => {
      return tWithoutNs(key, query, { ns: 'main', ...options })
    },
  } as { t: Translate })
```

**Comment by astorozhevsky** [comment]
> Our situation is a bit specific that we have several deployment, but each has only a single translation. Therefore we are able to copy only the necessary translation before starting the server into current
> 
> a simplified version is something like this
> 
> ```
> import transCore from 'next-translate/transCore'
> import currentLocale from 'locales/current/main.json'
> import { Translate } from 'next-translate'
> 
> const lang = process.env.LOCALE || 'en'
> 
> const tWithoutNs = transCore({
>   lang,
>   config: {
>     locales: [locale],
>     defaultLocale: locale,
>     defaultNS: 'main',
>     loader: false,
>     pages: {
>       '*': ['main'],
>     },
>   },
>   pluralRules: new Intl.PluralRules(lang),
>   allNamespaces: { main: currentLocale as any },
> })
> 
> export const getTranslate = () =>
>   ({
>     t: (key: string, query?: any, options = {}) => {
>       return tWithoutNs(key, query, { ns: 'main', ...options })
>     },
>   } as { t: Translate })
> ```

@jakubjanousek Thank you so much for sharing your code and approach. In the end, we switched to using [next-intl](https://next-intl-docs.vercel.app/) which solved our problems.
